tests/integration/executor/test_callbacks.py

```diff
@@ -41,6 +41,12 @@ def teardown_module():
     callbacks.unregister_all()
 
 
+@pytest.fixture(autouse=True)
+def mock_mount_unmount(mocker):
+    mocker.patch("craft_parts.utils.os_utils.mount")
+    mocker.patch("craft_parts.utils.os_utils.umount")
```

Comment:
Overlay operations require overlayfs mounting and umounting. We mock these in the regular unit and integration tests, but a new set of "privileged" tests will be added after the implementation is complete to actually mount overlay, perform chroot, etc.